src/ocamlorg_frontend/pages/exercises.eml

```diff
@@ -154,4 +160,47 @@ Learn_layout.three_column_layout
     <% exercises |> List.iter (fun (exercise : Data.Exercise.t) -> %>
     <%s! render_exercise exercise %>
     <% ); %>
-  </div>
+    <script>
+      document.addEventListener('alpine:init', () => {
+  
+        function computeSectionYPositions(toc_el) {
+          function get_y(href) {
+            let heading = document.getElementById(href.substring(1));
+            return heading.getBoundingClientRect().top + window.scrollY - 60;
+          }
+  
+          let sections = {};
+  
+          let els = toc_el.querySelectorAll("a");
+          let l = els.length;
+          for (let i=0; i<l; i++) {
+            let target_href = els[i].getAttribute("href");
+            let next_el_href = i < l-1 ? els[i+1].getAttribute("href") : null;
+  
+            sections[target_href] = {
+              start: get_y(target_href),
+              end: next_el_href ? get_y(next_el_href) : null,
+            }
+          }
+  
+          return sections;
+        }
+  
+        function isWithin(scrollY, href, sectionYPositions) {
+          if (sectionYPositions == null) return false;
+          return scrollY > sectionYPositions[href].start
+            && (scrollY <= sectionYPositions[href].end || sectionYPositions[href].end == null)
+        }
+  
+        Alpine.data('exercises', () => (
+          {
+            scrollY: window.scrollY,
+            sectionYPositions: null,
+  
+            isWithin,
+            computeSectionYPositions,
+          }
+        ))
+      })
+    </script>
+  </div>
```

Comment:
does this belong here? looks unrelated to dark mode

Reply:
yes it's unrelated but I noticed there's no active state on the exercise menu so I added it here.

Reply:
good idea. Would be better if this was a separate PR because then this feature addition is self contained

Reply:
yeah true. I'll create a separate pr for this then